trace-telemetry-exporter/src/main.rs

```diff
@@ -0,0 +1,185 @@
+use anyhow::Result;
+use chrono::{DateTime, Utc};
+use clap::Parser;
+use metrics::{counter, gauge};
+use metrics_exporter_prometheus::PrometheusBuilder;
+use rdkafka::{
+    consumer::{stream_consumer::StreamConsumer, CommitMode, Consumer},
+    message::Message,
+};
+use std::net::SocketAddr;
+use supermusr_streaming_types::dat1_digitizer_analog_trace_v1_generated::{
+    digitizer_analog_trace_message_buffer_has_identifier, root_as_digitizer_analog_trace_message,
+    DigitizerAnalogTraceMessage,
+};
+use tracing::{debug, trace, warn};
+
+const METRIC_MSG_COUNT: &str = "digitiser_message_received_count";
+const METRIC_LAST_MSG_TIMESTAMP: &str = "digitiser_last_message_timestamp";
+const METRIC_LAST_MSG_FRAME_NUMBER: &str = "digitiser_last_message_frame_number";
+const METRIC_CHANNEL_COUNT: &str = "digitiser_channel_count";
+const METRIC_SAMPLE_COUNT: &str = "digitiser_sample_count";
+
+#[derive(Debug, Parser)]
+#[clap(author, version, about)]
+struct Cli {
+    #[clap(long)]
+    broker: String,
+
+    #[clap(long)]
+    username: Option<String>,
+
+    #[clap(long)]
+    password: Option<String>,
+
+    #[clap(long = "group")]
+    consumer_group: String,
+
+    #[clap(long)]
+    trace_topic: String,
+
+    #[clap(long, env, default_value = "127.0.0.1:9091")]
+    metrics_address: SocketAddr,
+
+    #[clap(long, default_value_t = 5)]
+    message_rate_interval: u64,
+}
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    tracing_subscriber::fmt::init();
+
+    let args = Cli::parse();
+
+    let consumer: StreamConsumer = supermusr_common::generate_kafka_client_config(
+        &args.broker,
+        &args.username,
+        &args.password,
+    )
+    .set("group.id", &args.consumer_group)
+    .set("enable.partition.eof", "false")
+    .set("session.timeout.ms", "6000")
+    .set("enable.auto.commit", "false")
+    .create()?;
+
+    consumer.subscribe(&[&args.trace_topic])?;
+
+    let builder = PrometheusBuilder::new();
+    builder
+        .with_http_listener(args.metrics_address)
+        .install()
+        .expect("prometheus metrics exporter should be setup");
+
+    metrics::describe_counter!(
+        METRIC_MSG_COUNT,
+        metrics::Unit::Count,
+        "Number of messages received"
+    );
+
```

Comment:
Just realised I missed an obvious metric out of the initial list.

Message rate should also be included.